atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```
@@ -2129,53 +2126,60 @@ public void compactInternally(TableReference tableRef) {
             try {
                 clientPool.runOnHost(host, CassandraVerifier.healthCheck);
-                try {
-                    clientPool.runOnHost(host, clientPool.validatePartitioner);
-                } catch (Exception e) {
+                if (!partitionerIsValid(host)) {
                     return NodeAvailabilityStatus.TERMINAL;
                 }
             } catch (Exception e) {
                 numberOfUnreachableNodes++;
             }
         }
-        int quorumSize = replicationFactor / 2 + 1;
-        if (numberOfUnreachableNodes > (replicationFactor - quorumSize)) {
-            return NodeAvailabilityStatus.NO_QUORUM_AVAILABLE;
-        } else if (numberOfUnreachableNodes == 0) {
-            return NodeAvailabilityStatus.ALL_AVAILABLE;
-        } else {
-            return NodeAvailabilityStatus.QUORUM_AVAILABLE;
-        }
+        return getClusterAvailabilityStatus(numberOfUnreachableNodes);
     }
 
-    private void readAtSpecifiedConsistency(ConsistencyLevel consistency) throws TException {
-        clientPool.run(client -> {
-            String lockRowName = getHexEncodedBytes(CassandraConstants.GLOBAL_DDL_LOCK_ROW_NAME);
-            String lockColName = getHexEncodedBytes(CassandraConstants.GLOBAL_DDL_LOCK_COLUMN_NAME);
-            String selectCql = String.format(
-                    "SELECT \"value\" FROM \"%s\" WHERE key = %s AND column1 = %s AND column2 = -1;",
-                    schemaMutationLockTable.getOnlyTable().getQualifiedName(),
-                    lockRowName,
-                    lockColName);
-            return runCqlQuery(selectCql, client, consistency);
+    private boolean doesConfigReplicationFactorMatchWithCluster() {
+        return clientPool.run(client -> {
+            try {
+                CassandraVerifier.currentRfOnKeyspaceMatchesDesiredRf(client, configManager.getConfig(), false);
+                return true;
+            } catch (Exception e) {
+                return false;
```

For sure, you should log the exception message instead of just swallowing it.